src/lib/validation.ts

```diff
@@ -0,0 +1,432 @@
+					console.log(`\timage: ${inspect(i)}`);
+					return false;
+				}
+				return true;
+			},
+			commit: (c: any) => {
+				if (c != null && !isValidShortText(c)) {
+					console.log('debug: invalid commit passed to validation.isValidDependentAppsObject');
+					console.log(`\tcommit: ${inspect(c)}`);
+					return false;
+				}
+				return true;
+			},
+			config: (c: any) => {
+				if (!undefinedOrValidEnv(c)) {
+					console.log('debug; Invalid config passed to validation.isValidDependentAppsObject');
+					console.log(`\tconfig: ${inspect(c)}`);
+					return false;
+				}
+				return true;
+			},
+			environment: (e: any) => {
+				if (!undefinedOrValidEnv(e)) {
+					console.log('debug; Invalid environment passed to validation.isValidDependentAppsObject');
+					console.log(`\tenvironment: ${inspect(e)}`);
+					return false;
+				}
+				return true;
+			}
+		});
+	});
+}
+
+function isValidService(service: any, serviceId: string): boolean {
+	if (!(isValidShortText(serviceId) && checkInt(serviceId))) {
+		console.log('debug: Invalid service id passed to validation.isValidService');
+		console.log(`\tserviceId: ${inspect(serviceId)}`);
+		return false;
+	}
+
+	return _.conformsTo(service, {
+		serviceName: (n: any) => {
+			if (!isValidShortText(n)) {
+				console.log('debug: Invalid service name passed to validation.isValidService');
+				console.log(`\tserviceName: ${inspect(n)}`);
+				return false;
+			}
+			return true;
+		},
+		image: (i: any) => {
+			if (!isValidShortText(i)) {
+				console.log('debug: Invalid image passed to validation.isValidService');
+				console.log(`\timage: ${inspect(i)}`);
+				return false;
+			}
+			return true;
+		},
+		environment: (e: any) => {
+			if (!isValidEnv(e)) {
+				console.log('debug: Invalid env passed to validation.isValidService');
+				console.log(`\tenvironment: ${inspect(e)}`);
+				return false;
+			}
+			return true;
+		},
+		imageId: (i: any) => {
+			if (checkInt(i) == null) {
+				console.log('debug: Invalid image id passed to validation.isValidService');
+				console.log(`\timageId: ${inspect(i)}`);
+				return false;
+			}
+			return true;
+		},
+		labels: (l: any) => {
+			if (!isValidLabelsObject(l)) {
+				console.log('debug: Invalid labels object passed to validation.isValidService');
+				console.log(`\tlabels: ${inspect(l)}`);
+				return false;
+			}
+			return true;
+		}
+	});
+}
+
+/**
+ * isValidAppsObject
+ *
+ * Given an apps object from the state endpoint, validate the fields and
+ * return whether it's valid.
+ *
+ * TODO: Type the input correctly
+ */
+export function isValidAppsObject(obj: any): boolean {
+	if (!_.isObject(obj)) {
+		console.log('debug: Invalid object passed to validation.isValidAppsObject');
+		console.log(`\tobj: ${inspect(obj)}`);
+		return false;
+	}
+
+	return _.every(obj, (val, appId) => {
+		if (!(isValidShortText(appId) && checkInt(appId) != null)) {
```

Comment:
Can you refactor this to `if (!isValidShortText(serviceId) || !checkInt(serviceId)) {` ?